### PR TITLE
AMP-52 - Proof of concept deploy script

### PR DESCRIPTION
**Pull request by olibrook**
```
    Add deployment script for Lambda and static - not fit for prod

    - Uses AWS cloudformation to deploy two stacks:

      1. A stack shared by all AMP projects to configure S3 buckets
         which host code and static assets.
      2. A stack unique for each deployment of a Lambda, containing
         the Lambda and API config, etc.

    - Running the script requires and AWS account and revelent
      permissions on a developers machine.

    - Not fit for prod because we won't give parters access to Mobify's
      AWS account.

    - The plan is to have the deployment script run on mobify-cloud.
      This will be trivial to rewrite when the time comes, since we're
      using Cloudformation templates.
```

 **JIRA**: https://mobify.atlassian.net/browse/AMP-52

## How to test-drive this PR
- Come to my desk and we'll review it together. It requires AWS permission and is a little awkward to test!

### CODE REVIEW

**Comment by olibrook** [comment]
TODO: Follow best-practices for hosting static assets, outlined here - https://docs.google.com/document/d/1DDFSmr2G3jcIm6eAZZg2tPItufBQoVizLT6ZeHVmav0/edit#heading=h.hfwcc3xa5mki

File: amp/scripts/build/cloudformation.yaml [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                    Access-Control-Allow-Methods:
+                      type: string
+                    Access-Control-Allow-Headers:
+                      type: string
+              x-amazon-apigateway-integration:
+                responses:
+                  default:
+                    statusCode: 200
+                    responseParameters:
+                      method.response.header.Access-Control-Allow-Methods: "'DELETE,GET,HEAD,OPTIONS,PATCH,POST,PUT'"
```
Comment (kschmidtdev) [inline amp/scripts/build/cloudformation.yaml]: Do we care about anything other than `GET`?

File: amp/scripts/build/cloudformation.yaml [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                      method.response.header.Access-Control-Allow-Headers: "'Content-Type,Authorization,X-Amz-Date,X-Api-Key,X-Amz-Security-Token'"
+                      method.response.header.Access-Control-Allow-Origin: "'https://example.com'"
```
Comment (kschmidtdev) [inline amp/scripts/build/cloudformation.yaml]: Should this be a different domain? Or omitted altogether?

File: amp/scripts/build/cloudformation.yaml [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                passthroughBehavior: when_no_match
+                requestTemplates:
+                  application/json: "{\"statusCode\": 200}"
+                type: mock
```
Comment (kschmidtdev) [inline amp/scripts/build/cloudformation.yaml]: What does this mean 😊

**Pull request merged**